src/cli/commands/version.js

```diff
@@ -89,9 +89,11 @@ export async function setVersion(
   }
   invariant(newVersion, 'expected new version');
 
-  if (newVersion === pkg.version) {
-    throw new MessageError(reporter.lang('publishSame'));
-  }
+  // we dont do a check whether the new version is the same as the one in package.json
+  // because of two reasons (see issue #3011):
+  // - npm registry will complain anway when it already exist (leave it to them to validate that)
+  // - some open source libraries/projects increment version just before deployment where
+  //   the new version already has been bumped in package.json
```

Comment:
Not sure this comment is needed.

However, shouldn't we just return early when the package version is the same as the version currently registered? Instead of running lifecycle hooks etc. Not doing this could cause issues if, for example, a `version` lifecycle script was registering a mercurial tag.